### DISCUSSION TITLE
Documentation of INAV and missunderstandings, partly very confusing

### DISCUSSION THREAD

**Discussion by Flight-instructor** [iNavFlight/inav#6442] [Ideas]
## Current Behavior

Hello INAV team,

I am new into INAV and fighting with some parts of the documentation.
Let me give you an example:
I read, that if RTH is engaged, but the craft continues to get more far away from home-point, that at an additional distance of 500m+ an emergency-landing is executed. So far, so good.
Then I read something about setting up a landing speed. And wham, I am confused.
What is meant by that speed? is it an airspeed in flight direction, or a vertical speed?
This kind of confusion makes it very hard for beginners to understand, which they would understand if a very clearly wording would have been used.
I would suggest to separate horizontal speed and vertical speeds clearly by using  suitable words.
Suggestion:
Speed is always horizontal, except it is named '"vertical speed" or "rate of climb"(ROC) or "rate of descent"(ROD).
If we say " landing speed", it should always be an airspeed, so, horizontal.
If we mean the vertical landing speed of a multicopter, lets name it descent-rate or just "vertical landing speed", which, translated to a fixed wing means a rate of descent, whereas a landing speed to be set also would be helpfull, too, to prevent a stall.

What else confused me a lot?
Sometimes there is a wording called "rate-mode", but in the list of modes, there is no rate-mode. Instead we have manual, acro, angle-mode, RTH, Poshold and so on, but I cant find "rate mode" and so I cant understand, what is meant when the documentation says, choose a rate mode.

Elswhere in the documentation is the wording " Only if no mode is selected". but what is "no mode", if I switch of everything, the the OSD and my Lua-Telemetry says "manual". For me is that, manual mode is selected.
So my question: Is manual mode meant, when the documentation is speaking about " no mode selected", or do I something wrong.
If manual mode is meant, it should state " manual" and not "no mode". If not, I would like to know, how to switch everything of, so that there is  not a "manual" indication on my screen. ( I guess, manual, acro and angle are "no modes" whereas NAV cruise, horizon RTH and Poshold are Modes, or nav modes? when the docu says, if no mode is selcted, does that mean no nav-modes are selcted?) If so, the docu hould state " if no nav-mode is selected". Somehow confusing, or not, or.. only for me?

The ambigous wording is not only confusing beginners, but also might contain some danger to fail due to missunderstanding.

A lot of air-accidents had happen only due to ambiguous wording.
In earlier years a pilot reported "ready for takeoff", an other pilot understood xyzzyxf take off and thought he was cleared for takeoff. > consequence was an accident.
Today the word takeoff is only used, when there is a "cleared for takeoff" to be ready today means "ready for departure" instead of "ready for takeoff".

Perform a three-sixty or turn by 10° ALLWAYS means left turn. If not, it is named turn right by 10° or perform a three-sixty to the right" Not ambiguous. And more, a three-sixy ( full circle ) takes exactly two minutes time ( 3°/s turn-rate ) if not otherwise called, which happens more or less never. Everything in aviation is exactly given, without any doubt, to avoid missunderstandings.
For a fixed wing model, 2 mins is quiet a lot, but that is something to be changed. 
But, instead of giving a radius of the loiter-circles, It would be better to give a time for a full circle. The raduis is pending on the airspeed. With a Bixler, I can perform circles with a radius of 20 meters. Try this with a Jet at 400 km/h. But, both are able to perform a circle in a minute or even in 30s.  This is defined by an appropriate turn rate.

So, if I switch to PosHold, I expect the craft loitering to the left, otherwise, I am confused about the the FC has understood, what I pretented it to do. For a right turn, the command should be "PosHold turning right" or "loiter right". 

A lot of other examples could be named here, but I think thats enough to understand what I want to say.

What do I expect:
Nothing, because I know that documentation is an awfull lot of work and your time is really limited.

What could I do?

If you provide me with with docs, I could help to write documentations for end-users or, if I may ask questions, answered by the team, I could make suggestions for documentation updates or jeust write them and provide it back to team.
The documentation I mean is the "how to use INAV in your plane" docu, not such things for programmers as my knowledge about the s not enough to ask the right questions.

This where my first impressions with INAV and the documentation. And because there are still some missunderstandings, I refused from trying to fly with my matek in any other mode than manual. To be honest, I am angry to have something not understood right, failures in setting up something, and so on because in many places the docu is not clear, but ambiguous.

Call me an idiot, that I make suggestions here without having tried a flight with my FC in any modes other than manual. But as far as I do not understand what is going on, when I switch from a to b and why this is going on, I should not fly with a technique, unknown by me.
I would like to know in advance, what will happen, if my setup is so or so or so.
As of now, I only use the telemetry ( OSD ) and fly according to the artifical horizon, which works really good. I have no need for any nav-modes or such things. But this is individual. I go out to fly my plane, not to look its flying:-)) while doing nothing.

**Comment by DzikuVx** [comment]
@Flight-instructor INAV is open-source and as such any help is more than welcome. 
The INAV docs are either MD files based or stored in GitHub Wiki. You are free to edit both and submit to the INAV repository.
We would be grateful if you do

**Comment by digitalentity** [comment]
Thanks for the honest feedback @Flight-instructor, I really appreciate it.

INAV documentation is partly located in the [wiki](https://github.com/inavFlight/inav/wiki), partly in the [docs](https://github.com/iNavFlight/inav/tree/master/docs) folder. If you are willing to take on making a beginner-friendly manual, you're more than welcome to do so, we would be happy to clarify or explain stuff that's not clear.

**Comment by Flight-instructor** [comment]
Hi Pawel, Hi Konstantin,
    first of all, thanks for your quick reply.
    My Name is Ludger ( complete Ludger Kuhnen, but as Pawel allways
      says, just Pawel, ... ähm, Ludger.-)) )
    I just went into the wiki and had a look, how to edit the pages.
      I think, I found out, thats fine.
    And yes, I have questions, when reading the wiki.
    Let me give you an example, that astonished me:
    In the RTH setup, just in the first lines I read the following:
    Setting up Failsafe for RTH
    
      
    Foreword
    The goal is to configure both your flight controller and radio
      receiver so that failsafe does as you expect in every situation.
    For failsafe to work optimally iNav needs to know it's in a
      failsafe event and not just doing regular RTH. This is necessary
      for example to correctly handle loss of GPS while returning to
      home.
    This assumes you have regular GPS modes like RTH
      working already.
    
      
    Configuration of receiver
    You have two options on how to configure receiver:
    
      
    Option one
    Set receiver to send out NO PULSES or HOLD
      on a failsafe event. This is perfectly fine for FrSky Radios.
    
      I understand, that the FC will get into
          Failsafe ( mode as set ) when not receiving pulses, so far, so
          clear.
    But on Hold??? I imagine my plane is on NAV
        mode with heading and altitude controlled by FC. I look out my
        Aircraft ( by camera ) and I am luck not to have to do anything
        else, which means, 
      
    no stick inputs at all. Now, radioloink is
        lost, and my receiver goes on hold, which means it sends out the
        same pulses that it had before ( so, no change in behavior ).
        Why is that perfectly fine for FrSky radios ( I think, because
        they have the "hold pulses" function while others do not ), but
        how will my FC detect the radioloss, if I do so? I think, this
        is everything else than totally fine for whatever radio.
    As far as I understand, failsafe activated
        on hold could activate failsfe also, if there are no changing
        stick-commands for a time. That cannot be, I am sure, but ... 
      
    What is meant by totally fine for
        FrSky-Radios? 
      
    I set mine up in a way that it activates
        RTH-mode, which seems not to be the best way I think, but I am
        unsure. I will change my setup now to activate a failsafe-switch
        activating failsafe-RTH.
    
    In this case, as of now, I think, that it
        is a bad idea to set the pulses on hold. ( btw.I think, normally means, I am unsure )
    Option two ( this is totally clear for me
        )
    
    
      
        In Modes tab select a switch for "Failsafe"
      
      
        Set your switches and sticks of your radio to the following
          on link loss:
        
          Failsafe mode: activated
        
      
    
    The above is fine on FlySky radio.
    ______________________________________________
    
    
    Thats it for the moment.
    After your reply, I will edit the section to become
      understandable for everyone.
    PS: there had been questions also in my initial post.
    b.E.
    1. what is meant by "no mode" selected? Is that the case, if I am
      in manual mode? or can I even switch out manual mode, too? I did
      not find that somewhere.
    2. "Landing speed" at an emergency landing is vertical or
      horizontal speed? I guess its vertical for the multis.
    3. when I switch into any mode that is also holding my altitude,
      is that made by GPS-Alt or by baro, or can I choose?
    
    
    Answers to this questions could enable me to edit several doc´s
      in the documentation.
    Last but not least:
    Thank you both for working on this beautyfull software. 
    
    
      
        
    Am 22.12.2020 um 20:29 schrieb Paweł
      Spychalski:
    
    
      
      @Flight-instructor INAV is
        open-source and as such any help is more than welcome.
        The INAV docs are either MD files based or stored in GitHub
        Wiki. You are free to edit both and submit to the INAV
        repository.
        We would be grateful if you do
      —
        You are receiving this because you were mentioned.
        Reply to this email directly, view it on GitHub, or unsubscribe.
      [
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/iNavFlight/inav/discussions/6442#discussioncomment-235311",
"url": "https://github.com/iNavFlight/inav/discussions/6442#discussioncomment-235311",
"name": "View Discussion"
},
"description": "View this Discussion on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]
    
   
	
      
		Virenfrei. www.avast.com
